Userland/Libraries/LibCore/IODevice.cpp

```diff
@@ -108,14 +108,32 @@ bool IODevice::can_read_line() const
 {
     if (m_eof && !m_buffered_data.is_empty())
         return true;
+
     if (m_buffered_data.contains_slow('\n'))
         return true;
+
     if (!can_read_from_fd())
         return false;
-    populate_read_buffer();
-    if (m_eof && !m_buffered_data.is_empty())
-        return true;
-    return m_buffered_data.contains_slow('\n');
+
+    while (true) {
+        // populate buffer until a \n is found or we reach eof.
```

Comment:
Comments should be complete sentences:

```suggestion
        // Populate buffer until a newline is found or we reach EOF.
```

Reply:
done